modules/renter/contractor/update.go

```diff
@@ -0,0 +1,33 @@
+package contractor
+
+import (
+	"github.com/NebulousLabs/Sia/modules"
+	"github.com/NebulousLabs/Sia/types"
+)
+
+// renewThreshold calculates an appropriate renewThreshold for a given period.
+func renewThreshold(period types.BlockHeight) types.BlockHeight {
+	threshold := period / 4
+	if threshold > 1000 {
+		threshold = 1000
+	}
+	return threshold
+}
+
+// ProcessConsensusChange will be called by the consensus set every time there
+// is a change in the blockchain. Updates will always be called in order.
+func (c *Contractor) ProcessConsensusChange(cc modules.ConsensusChange) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if c.blockHeight != 0 || cc.AppliedBlocks[len(cc.AppliedBlocks)-1].ID() != types.GenesisBlock.ID() {
+		c.blockHeight += types.BlockHeight(len(cc.AppliedBlocks))
+		c.blockHeight -= types.BlockHeight(len(cc.RevertedBlocks))
+	}
```

Comment:
oh, the explorer probably uses a different trick. The explorer probably sets the initial height to -1 (underflowing it because it's unsigned), then lets the genesis block increment to 0. That works, but then you have an intentional underflow in your code and you need to do the underflow at initialization. I didn't think that was good code practice, so I phased it out in favor of the loops.